core/slice.cpp

```diff
@@ -26,10 +26,6 @@ void validate_begin(const scipp::index begin_) {
 Slice::Slice(const Dim dim_, const scipp::index begin_, const scipp::index end_)
     : m_dim(dim_), m_begin(begin_), m_end(end_) {
   validate_begin(begin_);
-  if (end_ != -1 && begin_ > end_)
-    throw except::SliceError("end must be >= begin. Given begin " +
-                             std::to_string(begin_) + " end " +
-                             std::to_string(end_));
```

Comment:
`end = -1` has a special meaning and implies a non-range slice, so I don't think you can just remove this check? Otherwise `-2` would mean empty slice, which would be really inconsistent, especially since we can't be sure that we don't have existing code that just checks for `end < 0` instead of `end == -1`.

No added tests, e.g., for using `transform` or methods of `VariableView`?

Have you considered to just fix this in the Python bindings, which should bring much less risk of breaking anything?